src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -616,6 +627,82 @@ private static OperationStatus DecodeWithWhiteSpaceFromUtf8InPlace(Span<byte> ut
             return status;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Avx512BW))]
+        [CompExactlyDependsOn(typeof(Avx512Vbmi))]
+        private static unsafe void Avx512Decode(ref byte* srcBytes, ref byte* destBytes, byte* srcEnd, int sourceLength, int destLength, byte* srcStart, byte* destStart)
+        {
+            // Reference for VBMI implementation : https://github.com/WojciechMula/base64simd/tree/master/decode
+            // If we have AVX512 support, pick off 64 bytes at a time for as long as we can,
+            // but make sure that we quit before seeing any == markers at the end of the
+            // string. Also, because we write 16 zeroes at the end of the output, ensure
+            // that there are at least 22 valid bytes of input data remaining to close the
+            // gap. 64 + 2 + 22 = 88 bytes.
+            byte* src = srcBytes;
+            byte* dest = destBytes;
+
+            // The JIT won't hoist these "constants", so help it
+            // JIT will remove the Non VBMI path constants when VBMI is available and vice-versa.
+            Vector512<sbyte> vbmiLookup0 = Vector512.Create(
+                0x80808080, 0x80808080, 0x80808080, 0x80808080,
+                0x80808080, 0x80808080, 0x80808080, 0x80808080,
+                0x80808080, 0x80808080, 0x3e808080, 0x3f808080,
+                0x37363534, 0x3b3a3938, 0x80803d3c, 0x80808080).AsSByte();
+            Vector512<sbyte> vbmiLookup1 = Vector512.Create(
+                0x02010080, 0x06050403, 0x0a090807, 0x0e0d0c0b,
+                0x1211100f, 0x16151413, 0x80191817, 0x80808080,
+                0x1c1b1a80, 0x201f1e1d, 0x24232221, 0x28272625,
+                0x2c2b2a29, 0x302f2e2d, 0x80333231, 0x80808080).AsSByte();
+            Vector512<byte> vbmiPackedLanesControl = Vector512.Create(
+                0x06000102, 0x090a0405, 0x0c0d0e08, 0x16101112,
+                0x191a1415, 0x1c1d1e18, 0x26202122, 0x292a2425,
+                0x2c2d2e28, 0x36303132, 0x393a3435, 0x3c3d3e38,
+                0x00000000, 0x00000000, 0x00000000, 0x00000000).AsByte();
+
+            Vector512<sbyte> mergeConstant0 = Vector512.Create(0x01400140).AsSByte();
+            Vector512<short> mergeConstant1 = Vector512.Create(0x00011000).AsInt16();
+
+            // This algorithm requires AVX512VBMI support.
+            // Vbmi was first introduced in CannonLake and is avaialable from IceLake on.
+            // This makes it okay to use Vbmi instructions since Vector512.IsHardwareAccelerated returns True only from IceLake on.
```

Comment:
This comment isn't quite accurate.

`Vector512.IsHardwareAccelerated` can be made to return `true` for `Skylake-X` and up to before `IceLake` via an environment variable. This is why the caller has the check for `Vector512.IsHardwareAccelerated && Avx512Vbmi.IsSupported` and why this function has `[CompExactlyDependsOn(typeof(Avx512Vbmi))]`

We're fine with it not being usable pre IceLake since those often incur heavier downclocking and its unnecessary complexity for a non-default scenario.

Reply:
Removed line 667